.dotnet/src/Custom/Chat/ChatClient.cs

```diff
@@ -99,14 +123,18 @@ public virtual Task<ClientResult<ChatCompletion>> CompleteChatAsync(string messa
         int choiceCount,
         ChatCompletionOptions options = null)
     {
-        Internal.Models.CreateChatCompletionRequest request = CreateInternalRequest(messages, options, choiceCount);
-        ClientResult<Internal.Models.CreateChatCompletionResponse> response = Shim.CreateChatCompletion(request);
+        Argument.AssertNotNull(messages, nameof(messages));
+        Internal.Models.CreateChatCompletionRequest internalRequest = CreateInternalRequest(messages, options, choiceCount);
+        using BinaryContent content = BinaryContent.Create(internalRequest);
+        ClientResult protocolResult = CompleteChat(content, OpenAIClient.DefaultRequestOptions);
```

Comment:
```suggestion
        ClientResult protocolResult = CompleteChat(content, options: null);
```

... I think this should work if we're on System.ClientModel v1.1.0.-beta.3.  If not, IIRC @joseharriaga had a workaround that included a null check inside the protocol method.

Reply:
I edited the generate code to include a null check, so passing null here is fine. 😊